docs/example_csharp.md

```diff
@@ -1,10 +1,256 @@
+
+namespace hello_tilt.Pages
+{
+    public class IndexModel : PageModel
+    {
+        private readonly ILogger<IndexModel> _logger;
+
+        public IndexModel(ILogger<IndexModel> logger)
+        {
+            _logger = logger;
+        }
+
+        public void OnGet()
+        {
+
+        }
+    }
+}
+```
+
+To start this server on Kubernetes, we need three configs:
+
+1. A [Dockerfile](https://github.com/windmilleng/tilt-example-csharp/blob/master/0-base/hello-tilt/Dockerfile) that builds the image
+2. A [Kubernetes deployment](https://github.com/windmilleng/tilt-example-csharp/blob/master/0-base/kubernetes.yaml) that runs the image
+3. And finally, a [Tiltfile](https://github.com/windmilleng/tilt-example-csharp/blob/master/0-base/Tiltfile) that ties them together:
+
+```python
+docker_build('hello-tilt', './hello-tilt')
+k8s_yaml('kubernetes.yaml')
+k8s_resource('hello-tilt', port_forwards='8080:80')
+```
+
+The first line tells Tilt to build an image with the name `hello-tilt` in the directory `./hello-tilt`.
+
+The second line tells Tilt to load the Kubernetes Deployment YAML. The image name in the `docker_build` call must match the container `image` reference in the `hello-tilt` Deployment.
+
+The last line configures port-forwarding so that your server is reachable at `localhost:8080`. The resource name in the `k8s_resource` call must match the Deployment's `metadata.name.
+
+Try it! Run:
+
+```
+git clone https://github.com/windmilleng/tilt-example-csharp
+cd tilt-example-csharp/0-base
+tilt up
+```
+
+Tilt will open a browser showing the web UI, a unified view that shows you app status and logs. Your terminal will also turn into a status box if you'd like to watch your server come up there.
+
+When the server is ready, you will see the status icon turn green. The log pane will display:
+```
+info: Microsoft.Hosting.Lifetime[0]
+      Now listening on: http://[::]:
+```
+
+{% assign case = "base" %}
+{% include example_guide_image.html %}
+
+## Step 1: Let's Add Benchmark Trickery
+
+Before we try to make this faster, let's measure it.
+
+Using [`local_resource`](local_house.html), you can direct Tilt to execute existing scripts or arbitrary shell commands on your own machine, and manage them from your sidebar like any other Tilt resource. We're going to use this functionality to benchmark our deploys.
+
+First we add a `local_resource` to our [Tiltfile](https://github.com/windmilleng/tilt-example-csharp/blob/master/1-measure/Tiltfile) that records the start time in a C# file.
+
+```python
+local_resource(
+    'deploy',
+    './record-start-time.sh',
+    deps=['./record-start-time.sh']
+)
+```
+
+The `local_resource()` call creates a local resource named `deploy`. The second argument is the script that it runs.
+
+We've also modified our server to read that start itme, calculate the time elapsed, then display this in the rendered HTML.
+
+Let's click the button on the `deploy` resource and see what happens!
+
+<figure>
```

Comment:
you should probably use templating for all of your images -- but i'm applying review comments on #488 and how we do this will change slightly, so stand by (or else merge as is and i'll mix as part of my PR)